streamer/src/nonblocking/quic.rs

```diff
@@ -1815,7 +1811,7 @@ pub mod test {
         join_handle.await.unwrap();
     }
 
-    #[tokio::test]
+    #[tokio::test(flavor = "multi_thread")]
```

Comment:
By default tokio::test uses the single threaded runtime. This is why the huge timeout was needed, because if things were scheduled so that the thread went inside epoll_wait(), then it'd get stuck in there for however long the timeout is, messing with the (fragile) timing assumptions made by the test.